rust_dev_preview/sitewise/src/bin/list-assets.rs

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use aws_config::meta::region::RegionProviderChain;
+use aws_sdk_iotsitewise::{model::ListAssetsFilter, Client, Error, Region, PKG_VERSION};
+use aws_smithy_types_convert::date_time::DateTimeExt;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+struct Opt {
+    /// The AWS Region.
+    #[structopt(short, long)]
+    region: Option<String>,
+
+    /// The asset model id.
+    #[structopt(short, long)]
+    asset_model_id: Option<String>,
+
+    /// The filter.
+    #[structopt(short, long)]
+    filter: String,
+
+    /// Whether to display additional information.
+    #[structopt(short, long)]
+    verbose: bool,
+}
+
+// List the assets under IoT SiteWise.
+// snippet-start:[sitewise.rust.list-assets]
+async fn list_assets(
+    client: &Client,
+    filter: ListAssetsFilter,
+    asset_model_id: Option<String>,
+) -> Result<(), Error> {
+    let resp = client
+        .list_assets()
+        .filter(filter)
+        .set_asset_model_id(asset_model_id)
+        .send()
+        .await?;
+
+    println!("Assets:");
+
+    for asset in resp.asset_summaries.unwrap() {
+        println!("  ID:  {}", asset.id.as_deref().unwrap_or_default());
+        println!("  ARN:  {}", asset.arn.as_deref().unwrap_or_default());
+        println!("  Name:   {}", asset.name.as_deref().unwrap_or_default());
+        println!(
+            "  Asset Model ID:   {}",
+            asset.asset_model_id.as_deref().unwrap_or_default()
+        );
+        println!(
+            "  Creation Date:   {}",
+            asset.creation_date.unwrap().to_chrono_utc()
+        );
+        println!(
+            "  Last Update Date:   {}",
+            asset.last_update_date.unwrap().to_chrono_utc()
+        );
+        println!(
+            "  Current Status:   {}",
+            asset.status.unwrap().state.unwrap().as_str()
+        );
+
+        println!("  Assets Hierarchies:");
+
+        for hierarchy in asset.hierarchies.unwrap() {
+            println!(
+                "    Hierarchy ID:   {}",
+                hierarchy.id.as_deref().unwrap_or_default()
+            );
+            println!(
+                "    Hierarchy Name:   {}",
+                hierarchy.name.as_deref().unwrap_or_default()
+            );
+        }
+
+        println!();
+    }
+
+    println!();
+
+    Ok(())
+}
+// snippet-end:[sitewise.rust.list-assets]
+
+/// Lists the id, arn, name, asset_model_id, creation_date, last_update_data, status
+/// and hierarchies of your Sitewise Assets in the Region.
```

Comment:
Lists the ID, Amazon Resource Name (ARN), name, asset_model_id, creation_date, last_update_data, status,
and hierarchies of your AWS IoT SiteWise assets in the Region.

Reply:
updated